docs/externalnode.md

```diff
@@ -0,0 +1,309 @@
+will be used to represent such a virtual machine or a bare-metal server in this
+document.
+
+`ExternalEntity` is used to represent the network interface defined in an
+`ExternalNode` object, when applying an Antrea NetworkPolicy to it.
+antrea-controller creates an `ExternalEntity` for each network interface when 
+it finds an `ExternalNode` is created, which can be used when applying Antrea
+NetworkPolicy.
+
+antrea-agent is running on the external Node which is described as `ExternalNode`,
+and controls its network connectivity and security of by attaching the network
+interface on OVS. A new OpenFlow pipeline is implemented dedicated for ExternalNode
+feature.
+
+You may be interested in using this capability if any of the following apply:
+
+- Antrea NetworkPolicy is wanted to provide security control on an external Node.
+
+- You want the same security configurations on the external Node, with no differences
+  on the Operating Systems installed on the machine.
+
+This guide demonstrates how to configure `ExternalNode` to achieve the above result.
+
+## Prerequisites
+
+ExternalNode is introduced in v1.8 as an alpha feature. A feature gate `ExternalNode`
+must be enabled on the antrea-controller and antrea-agent in the configuration.
+The configuration for antrea-controller is modified in the `antrea-config` ConfigMap
+like the following options for the feature to work:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: antrea-config-dcfb6k2hkm
+  namespace: kube-system
+data:
+  antrea-controller.conf: |
+    featureGates:
+      ExternalNode: true
+```
+
+Since antrea-agent is running on an external Node which is not managed by Kubernetes,
+a single configuration file is required to locate on each machine where antrea-agent
+is running. The following options for the feature are needed:
+```yaml
+featureGates:
+  ExternalNode: true
+nodeType: externalNode
+```
+Other options for antrea-agent running on an external Node please refer to the sample
+configuration [antrea-agent.conf](../build/yamls/externalnode/conf/antrea-agent.conf).
+
+## The ExternalNode resource
+
+A typical ExternalNode resource example:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ExternalNode
+metadata:
+  name: vm1
+  namespace: vm-ns
+  labels:
+    role: db
+spec:
+  interfaces:
+    - ips: ["172.16.100.3"]
+      name: ""
+```
+
+**Note**: `ExternalNode` resource is working in `namespace` scope.
+
+### Interfaces
+
+The `interfaces` field specifies the list of the network interfaces expected to be
+guarded by Antrea NetworkPolicy. One interface at least is required. The `name` or
+`ips` can be used to identify the target interface. **The field `ips` must be
+provided in the CRD**, but `name` is optional. Multiple IPs on a single interface is
+supported.
+
+A single `ExtrenalEntity` is created mapping to one interface, and antrea-controller
+performs the convention when it finds an `ExternalNode` is created.
```

Comment:
convertion

Reply:
mapping to one interface -> what is it mapping here. Can we add it in the same sentence

Reply:
Remove the words "mapping to". The relations/conversions between ExternalEntity and ExternalNode is listed.